### PR TITLE
Add source and fetcher for Docker containers

### LINKED ISSUE #33: Feature request: Add docker image support
There is [support for pulling docker images in nixpkgs](https://nixos.org/manual/nixpkgs/stable/#ssec-pkgs-dockerTools-fetchFromRegistry), and it would be awesome to use nvfetcher to update the digest/sha for this.

[nvchecker](https://nvchecker.readthedocs.io/en/latest/usage.html#check-container-registry) also has support for checking docker registries.

It seems like most of the pieces are in place, it just needs some wiring for nvfetcher to create sources. I've tried patching this together myself, but I'll probably need a crash course on haskell before I can actually do this, so I'll leave this issue here for now :)

### PR DESCRIPTION
**Pull request by amesgen**
Closes #33 

Supports new entries like
```toml
[alpine]
src.container = "library/alpine"
src.include_regex = "3\\..*"
fetch.docker = "library/alpine"
```
Additional options include
```toml
docker.arch = "arm64"
docker.tlsVerify = false
```

Adds a dependency on [`nix-prefetch-docker`](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/build-support/docker/nix-prefetch-docker.nix).

### CODE REVIEW

**Review by berberman — APPROVED** [review]
LGTM, Thank you!

File: test/Utils.hs [hunk 1]
```diff
@@ -87,3 +87,19 @@ specifyChan s m = specify s $ \r -> runReaderT m r
...
+    { _imageName = "library/alpine",
+      _imageTag = "3.16.2",
+      _imageDigest = (),
+      _sha256 = (),
+      _fos = Nothing,
+      _farch = Nothing,
+      _finalImageName = Nothing,
+      _finalImageTag = Nothing,
+      _tlsVerify = Nothing
+    }
```
[the remark this answers is not in the document]
Comment (amesgen) [reply test/Utils.hs:105]: Note that it is also used on `NixExprSpec`:

https://github.com/berberman/nvfetcher/blob/0d56b034c10ff516480f325436e54562897e3dee/test/NixExprSpec.hs#L73

Should I duplicate it?

[the remark this answers is not in the document]
Comment (berberman) [reply test/Utils.hs:105]: I would do that, as it's not related to the Util for test session :)

[the remark this answers is not in the document]
Comment (amesgen) [reply test/Utils.hs]: 7c65721

**Pull request merged**